src/fastoad/openmdao/variables/variable_list.py

```diff
@@ -326,13 +323,19 @@ def from_problem(
         # Manage variable promotion
         if not get_promoted_names:
             final_inputs = inputs
+
             final_outputs = outputs
         else:
             final_inputs = {
                 metadata["prom_name"]: dict(metadata, is_input=True) for metadata in inputs.values()
             }
             final_outputs = cls._get_promoted_outputs(outputs)
 
+            # Remove possible duplicates due to Indeps
+            for input_name in final_inputs:
+                if input_name in final_outputs:
+                    del final_outputs[input_name]
```

Comment:
I wouldn't have mentioned that if there was no other change to apply, but here, I would recommend:
```suggestion
            for var_name in set(final_inputs).intersection(final_outputs):
                del final_outputs[var_name]
```